experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,505 @@
+#This code uses Derksen's algorithm to compute fundamental invariants of linearly reductive group.
+#We first set up LinearlyReductiveGroup , then we set up a representation via which the group acts on a vector space (RepresentationLinearlyReductiveGroup)
+#Then we set up the invariant ring of the group. The fundamental invariants are computed using Derksen's alg.
+#As of now, the only reynolds operator that is implemented is the one for SLm, using Cayley's Omega process.
+
+export LinearlyReductiveGroup, reductive_group, representation_matrix, group, reynolds_operator, group_ideal, canonical_representation, natural_representation
+export RepresentationLinearlyReductiveGroup, representation_reductive_group, representation_on_forms, representation_matrix
+export RedGrpInvRing, invariant_ring, fundamental_invariants, null_cone_ideal
+##########################
+#Setting up Reductive Groups 
+##########################
+#These are objects that carry information about a linearly reductive group. 
+#As of now it is only implemented for SLn, its direct products and tensors. 
+
+mutable struct LinearlyReductiveGroup
+    field::Field #characteristic zero. implement check? 
+    group::Tuple{Symbol, Int}
+    group_ideal::MPolyIdeal
+    reynolds_operator::Function
+    canonical_representation::MatElem
+
+    function LinearlyReductiveGroup(sym::Symbol, m::Int, fld::Field) #have not decided the representation yet
+        #check char(fld)
+        @assert sym == :SL 
+        R, _ = polynomial_ring(fld, :z => (1:m,1:m))
+        return LinearlyReductiveGroup(sym, m, R)
+    end
+
+    function LinearlyReductiveGroup(sym::Symbol, m::Int, pring::MPolyRing) #the ring input is the group ring
+        #check char(field)
+        G = new()
+        @assert sym == :SL 
+        fld = base_ring(pring)
+        characteristic(fld) == 0 || error("Characteristic should be 0 for linearly reductive groups")
+        G.field = fld
+        @req m^2  == ngens(pring)
```

Comment:
```suggestion
        @req m^2  == ngens(pring) "some error message"
```
`@req` always needs an error message